crates/common_utils/Cargo.toml

```diff
@@ -8,16 +8,17 @@ readme = "README.md"
 license.workspace = true
 
 [features]
-signals = ["dep:signal-hook-tokio", "dep:signal-hook", "dep:tokio", "dep:router_env", "dep:futures"]
-async_ext = ["dep:futures", "dep:async-trait"]
+signals = ["dep:signal-hook-tokio", "dep:signal-hook", "dep:tokio", "dep:router_env"]
+async_ext = ["dep:async-trait"]
 logs = ["dep:router_env"]
+metrics = ["dep:router_env"]
 
 [dependencies]
 async-trait = { version = "0.1.79", optional = true }
 bytes = "1.6.0"
 diesel = "2.1.5"
 error-stack = "0.4.1"
-futures = { version = "0.3.30", optional = true }
+futures = { version = "0.3.30" }
```

Comment:
Can we make this dependency optional as well, and enable it for the `metrics` feature flag?